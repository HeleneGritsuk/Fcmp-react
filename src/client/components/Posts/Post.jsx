import React from 'react';
import styles from '../../styles/postsList.css.js';

class Post extends React.Component {

  render() {
    const {  author,  text,   title } = this.props;

    return (
      <div style = {styles.postContainer}>
        <div style = {styles.titleStyle}>{title}</div>
        <div style = {styles.authorWrapper}>by {author}</div>
        <div>Text: {text}</div>
      </div>
    );
  }
}
export default Post;
