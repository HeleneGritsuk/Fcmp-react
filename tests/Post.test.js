import React from 'react';
import Post from '../src/client/components/Posts/Post.jsx';
import renderer from 'react-test-renderer';

const mockPost = {
    author: "THE ASSOCIATED PRESS",
    id: "5ac1134b9bc42c482c71e594",
    text: "China's defunct Tiangong 1 space station is hurtling toward Earth and expected to re-enter the atmosphere within the next day.\nMost of it should burn up on re-entry, so scientists say it poses only a â€¦",
    title: "Defunct Chinese space lab set to re-enter Earth's atmosphere1"
  };

describe('Post', () => {
  it('should render Post content correctly', () => {
    const tree = renderer.create(<Post key={mockPost.id} {...mockPost} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
