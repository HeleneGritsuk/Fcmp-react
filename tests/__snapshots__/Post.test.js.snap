// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render Post content correctly 1`] = `
<div
  style={
    Object {
      "border": "1px solid #dadada",
      "borderRadius": "5px",
      "marginBottom": "15px",
      "padding": "15px",
      "width": "300px",
    }
  }
>
  <div
    style={
      Object {
        "fontWeight": "bold",
        "textTransform": "uppercase",
      }
    }
  >
    Defunct Chinese space lab set to re-enter Earth's atmosphere1
  </div>
  <div
    style={
      Object {
        "color": "#8d8d8d",
        "fontSize": "12px",
        "fontStyle": "italic",
        "textAlign": "right",
      }
    }
  >
    by 
    THE ASSOCIATED PRESS
  </div>
  <div>
    Text: 
    China's defunct Tiangong 1 space station is hurtling toward Earth and expected to re-enter the atmosphere within the next day.
Most of it should burn up on re-entry, so scientists say it poses only a â€¦
  </div>
</div>
`;
